src/main/java/seedu/duke/parser/Parser.java

```diff
@@ -0,0 +1,24 @@
+package seedu.duke.parser;
+
+import seedu.duke.commands.Command;
+import seedu.duke.commands.CommandType;
+import seedu.duke.commands.ExitCommand;
+import seedu.duke.commands.ShowRateCommand;
+public class Parser {
+    public static Command parseInput(String input) throws IndexOutOfBoundsException, IllegalArgumentException{
+        CommandType command;
+        String[] args = input.split(" ");
+
+        // trim whitespace and convert to uppercase so input matches the enum names
+        command = CommandType.get(args[0]);
```

Comment:
Consider declaring the variable when you need it as part of good coding standards.

Reply:
In the latest revision, the CommandType.get is wrapped in a try-catch block so it is necessary to declare the variable first